File: esthesis-platform-server/src/main/java/esthesis/platform/server/dto/DataSinkFactoryDTO.java
package esthesis.platform.server.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@Builder
public class DataSinkFactoryDTO {
  private String factoryClass;
  private String friendlyName;
  private boolean supportsMetadataRead;
  private boolean supportsTelemetryRead;
  private boolean supportsMetadataWrite;
  private boolean supportsTelemetryWrite;
  private String version;
  private String configurationTemplate;
}

File: esthesis-platform-ui/src/app/dto/data-sink-factory-dto.ts
export class DataSinkFactoryDto {
  factoryClass: string;
  friendlyName: string;
  supportsMetadataRead: boolean;
  supportsTelemetryRead: boolean;
  supportsMetadataWrite: boolean;
  supportsTelemetryWrite: boolean;
  version: string;
  configurationTemplate: string;
}
